src/types.rs

```diff
@@ -294,23 +312,40 @@ impl Note {
 
 impl core::fmt::Debug for Note {
     fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
-        write!(f, "a")
+        write!(f, "Note(\n")?;
```

Comment:
Especially if we implement Debug to be string instead of array, I would to implement `Debug` for Note (since it should contains anyhow values that implements already `Debug`).